presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -509,8 +509,6 @@ public synchronized void cancel()
                 return;
             }
 
-            checkState(taskStatusFetcher.isRunning(), "Cannot cancel task when it is not running");
```

Comment:
I am not sure why we had this, but we should be able to send a delete to a task (on the worker) even if it was not started (on the coordinator)